lib/streamlit/elements/lib/utils.py

```diff
@@ -175,9 +181,42 @@ def compute_and_register_element_id(
     The element ID gets registered to make sure that only one ID and user-specified
```

Comment:
I would add here that the id gets registered only if a `script_run_context` exists

Reply:
> I would add here that the id gets registered only if a script_run_context exists

Hmm, but I don't think we do this currently in any other situation where we check for context?

Reply:
yeah that's true but I think this function is a little bit special because it says "compute_and_register" where it always computes but only sometimes register. But no strong opinion